streaming/base/dataset.py

```diff
@@ -277,6 +323,14 @@ def __init__(self,
         self.stream_per_shard = np.array(stream_per_shard, np.int64)
         self.num_shards = len(self.shards)
 
+        # Check that cache limit is possible.
+        if self.cache_limit:
+            min_cache_usage = sum(map(lambda stream: stream.get_index_size(), streams))
```

Comment:
Isn't the min actually this size + the size of one shard? I guess that is harder to figure out up front?

Reply:
nvm, found the exception that happens in this case